src/main/java/com/patsi/service/RecipeService.java

```diff
@@ -29,37 +29,37 @@ public class RecipeService {
     @Value("${spring.web.resources.static-locations[3]}")
     private String IMAGE_PATH;
 
+    //Get Existing Recipe
+    public List<Recipe> getRecipe(String uid) {
+        List<Recipe> finalRecipeList = recipeRepository.findAll().stream()
+            .filter(recipe -> recipe.getUid() != null && recipe.getUid().equals(uid))
+            .collect(Collectors.toList());
+        finalRecipeList.forEach((recipe -> {
+            recipe.setUid(null);
+        }));
+        return finalRecipeList;
+    }
+
     //Register New Recipe
-    public UUID registerRecipe(Recipe recipe) {
+    public UUID registerRecipe(Recipe recipe, String userUid) {
+        recipe.setUid(userUid);
         List<Recipe.Ingredient> ingredients = recipe.getIngredient();
         recipe.setIngredient(ingredients);
         recipeRepository.save(recipe);
         return recipe.getRecipeID();
     }
 
-    //Update Existing Recipe
-    public void updateRecipe(UUID id, Recipe recipe) {
-        recipeRepository.save(recipe);
+    //Add Recipe Icon to folder before assigning to user
+    public void updateRecipeIcon(String id, byte[] recipeIcon) throws IOException {
+        log.info("In Service: updateRecipeIcon");
+        FileHelper.newFile(IMAGE_PATH, id, recipeIcon);
     }
 
-    public void updateRecipeIcon(UUID id, byte[] recipeIcon) throws IOException {
-        FileHelper.newFile(IMAGE_PATH, id.toString(), recipeIcon);
-        Recipe recipe = recipeRepository.findById(id).get();
-        recipe.setImgURL(id.toString());
+    //Update Existing Recipe
+    public void updateRecipe(UUID id, Recipe recipe) {
```

Comment:
Need validation on the recipe,  e.g. owner of the recipe, any mandatory field is missing, etc...

Can raise a ticket and implement it after refinement

Reply:
created ticket for this 
Link: https://smart-home-cloud.atlassian.net/browse/RECIPE-50